include/arch/arm/aarch64/timer.h

```diff
@@ -36,10 +37,13 @@ static ALWAYS_INLINE void arm_arch_timer_enable(unsigned char enable)
 	__asm__ volatile("mrs %0, cntv_ctl_el0\n\t"
 			 : "=r" (cntv_ctl) :  : "memory");
 
-	if (enable)
-		cntv_ctl |= CNTV_CTL_ENABLE;
-	else
-		cntv_ctl &= ~CNTV_CTL_ENABLE;
+	if (enable) {
+		cntv_ctl |= CNTV_CTL_ENABLE_MASK;
+		cntv_ctl &= ~CNTV_CTL_IT_MASK;
```

Comment:
Can you evaluate the fix proposed in https://github.com/zephyrproject-rtos/zephyr/pull/30164/commits/be5244b61d33e481320faa77a8a00f3886ab6ca4 ?

Reply:
ok, that fix seems better.